src/main/java/com/example/onlinebookstore/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package com.example.onlinebookstore.security;
+
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find by user name"));
```

Comment:
The conditions for this task contain the following item:

Add Liquibase changesets to insert roles into DB and assign role for existing users. Be sure you have at least one user in DB with role admin

This means that at the time of making new changes to the DB structure, the users table already contains user records, and therefore, at the time of making them, the password was not saved as a hash value. As for me, in addition to updating the table structures at this stage, it is necessary to bring the storage format of existing passwords into line. I decided to do this through liquibase. As for me, this option is more correct than manually converting passwords to their corresponding values ​​and then hardcoding them in scripts, since with a large number of existing users this will be tedious.